src/java.net.http/share/classes/jdk/internal/net/http/Http2Connection.java

```diff
@@ -181,6 +184,28 @@ class Http2Connection  {
      *  and SSLTube.recycler.
      */
 
+    // An Idle connection is one that has no active streams
+    // and has not sent the final stream flag
+    final class IdleConnectionTimeoutEvent extends TimeoutEvent {
+
+        IdleConnectionTimeoutEvent(Duration duration) {
+            super(duration);
+        }
+
+        @Override
+        public void handle() {
+            if (debug.on()) {
+                debug.log("HTTP connection idle for too long");
+            }
+            HttpTimeoutException hte = new HttpTimeoutException("HTTP connection idle, no active streams. Shutting down.");
+            shutdown(hte);
```

Comment:
The implementation in `shutdown` has some code which logs errors/exception when `error` logging is enabled. Specifically:

```
if (Log.errors()) {
            if (!(t instanceof EOFException) || isActive()) {
                Log.logError(t);
```
The implementation of `Log.logError` logs the exception stacktrace as well as a message which says `ERROR:`. Should we add a specific check in the `shutdown` implementation to prevent logging exception stacktraces when a connection is being closed due to idle timeout and instead just log it as an informational message?

Reply:
Good catch - yes we don't want the exception to be logged as an error if we're closing an idle connection.